docs/developer-guide.md

```diff
@@ -201,16 +201,42 @@ mvn clean compile checkstyle:check spotbugs:check
 
 ### Javadoc
 
-We try to ensure that all classes have Javadoc. Most methods should also have Javadoc. Private methods, as well as
-getters and setters can be skipped unless there's something important to know.
+We try to ensure that all production classes have Javadoc. Most methods should also have Javadoc. Javadoc should
```

Comment:
Not sure what 'production' means here?

Reply:
I've removed this, and moved up the explanation about test code.